packages/loader/container-utils/src/error.ts

```diff
@@ -87,6 +87,7 @@ export class ThrottlingWarning extends LoggingError implements IThrottlingWarnin
     constructor(
         message: string,
         readonly retryAfterSeconds: number,
+        public readonly originalError: unknown, // for logging
```

Comment:
i think the new param should go last, as i think unknown and ITelemetryProperties overload, so you wouldn't get a compile time error if there was an existing usage, but it would now be wrong

Reply:
in general, new optional params should always go last, i also dislike functions that take lots of optional params, and usually prefer moving to an object if more than a couple. probably not necessary here, but thought i'd share in case it comes up elsewhere

Reply:
It can't go last because it's optional, but I can put it first which will break if someone misses this change

Reply:
I'm also a little concerned about this since it crosses compat boundaries (loader <- runtime).

To be correctly backwards compatible, this new parameter needs to be optional (or at least handled as such). If you want to type-safely ensure that future uses always provide it, then I'd recommend adding a thin layer of abstraction over the raw constructor, like a static create function, and in work towards moving all callers to use that instead. Then, in the future, we can make the constructor itself private, giving more flexibility for transitioning creation of the object in the future (if we think it will be recurring). An alternative, and probably more practical solution here, would be to use an object for the parameters.

Additionally, I think for compatibility sake, it _needs_ to be the last parameter. In the case of new versions of loader interacting with old versions of runtime, they will either not provide this argument (so we need to be expecting undefined- which I think we already are), or they will provide props which will incorrectly go in as originalError.

___

After looking at the actual use cases in code, I see that we never call the constructor outside of this package directly, and instead already go through the `wrap` function only. In this case, I might go a step further and go ahead and make this constructor private or switching to an object for params to protect ourselves going forward.